projectDocs/dev/codingStandards.md

```diff
@@ -159,6 +159,69 @@ Type hints make reasoning about code much easier, and allow static analysis tool
 * Prefer union shorthand (`X | Y`) over explicitly using `typing.Union`.
   * Corollary: prefer `T | None` over `typing.Optional[T]`.
 
+### Calling non-python code
+
+Todo: Merge with section in #18207
+
+#### using the `ctypesUtils` module
+
+The goal of the `utils.ctypesUtils` module is to make calling external C functions easier.
+It does so by aiding in two areas:
+
+1. Annotating the ctypes C function pointer with type information (restype and argtypes) so that parameters passed to the function are properly validated by ctypes.
+2. Providing type hints to the several parameters of the function.
+
+The decorator is best explained by a basic example.
+
+```python
+from ctypes import windll
+from ctypes.wintypes import BOOL, HANDLE
+from typing import Annotated
+from utils.ctypesUtils import dllFunc
+
+@dllFunc(windll.kernel32)
+def CloseHandle(hObject: int | HANDLE) -> Annotated[int, BOOL]:
+	...
+```
+
+A properly annotated function is a function of the form:
+`def FunctionNameINDll(param1: hint1, param2: hint2) -> ReturnHint:`
+
+* By default, the `dllFunc` decorator infers the function name from the name of the Python function. It can be overridden by passing an additional parameter to the decorator.
```

Comment:
Please move new sentences to new lines
```suggestion
* By default, the `dllFunc` decorator infers the function name from the name of the Python function.
It can be overridden by passing an additional parameter to the decorator.
```